KeyDistribution_E91/Tasks.qs

```diff
@@ -0,0 +1,178 @@
+
+
+
+    //////////////////////////////////////////////////////////////////
+    // Part I. Preparation
+    //////////////////////////////////////////////////////////////////
+	
+    // Task 1.1. Entangled Pairs
+    // Inputs:
+    //      1) "qsAlice": an array of N qubits in the |0⟩ states,
+    //	    2) "qsBob": an array of N qubits in the |0⟩ states,
+    // Goal: Create entangled pairs |Ψ⟩ = 1/√2 (|00⟩ + |11⟩) out of corresponding qubits of
+    //        Alice's and Bob's qubits.
+    operation EntangledPairs (qsAlice : Qubit[], qsBob : Qubit[]) : Unit {
+        Fact(Length(qsAlice) == Length(qsBob), "Alice and Bob should have the same number of qubits");
+        // ...
+
+    }
+
+
+    //////////////////////////////////////////////////////////////////
+    // Part II. E91 Protocol
+    //////////////////////////////////////////////////////////////////
+
+    // Task 2.1 Rotate and Measure
+    // One of the essential steps of E91 protocol is conducting measurement operations.
+    // In Z-X plane, Alice measures her qubit in either one of Z, (Z+X)/√2 or X basis whereas
+    // Bob measures his qubit in either one of (Z+X)/√2, X or (X-Z)/√2 basis. 
+    // Inputs:
+    //     1) "q": the qubit to be measured,
+    //     2) "rotationMultiplier": an integer to me multiplied by π/4 to  
+    //                              compute the rotation angle in Z-X plane,
+    // Output: Result of the measurement
+    // Goal: Rotate PauliZ basis and measure q in that new basis.
+    operation RotateAndMeasure(q : Qubit, rotationMultiplier: Int) : Result {
+        // ...
+
+        return Zero;
+    }
+
+    // Task 2.2 Measure Qubit Arrays
+    // Now that you have a way of measuring a qubit in a given direction, it's time to 
+    // measure either Alice's or Bob's qubits in random bases selected from their respective
+    // basis sets.
+    // Inputs:
+    //     1) "qs": An array of qubits to be measured,
+    //     2) "basisListt": A list of 3 possible rotation mutlipliers.
+    // Outputs:
+    //     1) List of generated basis choices,
+    //     2) List of measurement results.
+    // Goal: Measure qubits from qs in bases constructed using rotation multipliers that
+    //       are choosen randomly from basisList.
+    operation MeasureQubitArray (qs: Qubit[], basisList: Int[]) : (Int[], Result[]) {
+        let results = new Result[0];
+        let bases = new Int[0];
+        // ...
+
+        return (bases, results);
+
+    }
+
+    // Task 2.3 Generate the shared key
+    // After both Alice and Bob make their measurements and share their choices of bases, it
+    // is possible to select compatible measurements and create a key out of these. You 
+    // should compute that key for either Alice or Bob.
+    // Inputs:
+    //     1) "basesAlice": Alice's measurement bases multipliers,
+    //     2) "basesBob": Bob's measurement bases multipliers,
+    //     3) "results": Either Alice's or Bob's measurement outcomes.
```

Comment:
If I understand correctly, Alice's and Bob's results with matching bases are going to be different; in this case we should say explicitly which ones we provide as a result. Might add a Bool parameter `alicesResults` which is true if results are from Alice and false if they are from Bob, so that the solution needs to be careful about interpreting the results.